### DISCUSSION TITLE
"visual block selector" for rich-text "embedded" components

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#3345] [Ideas]
Requested here: https://twitter.com/creativiii/status/1586093628325249035

Similar to our [visual blocks selector](https://tina.io/docs/editing/blocks/#adding-a-visual-block-selector-experimental) for the object field, this user would like a similar UI for an editor goes to embed a component in the rich-text field

**Comment by ghost** [comment]
Hi jamespohalloran.

This block selector feature for embedded components is what I initially requested - because one of my features would be to integrate tinacms with editor-js. Please see this: [Is it possible to use editor.js with tinicms?](https://github.com/tinacms/tinacms/discussions/3196) *- so you commented the other day.*

*img-before*
![image](https://user-images.githubusercontent.com/111701513/215019549-4d1c8357-db27-49e9-a5f8-d56af7ede2f4.png)

*img-after*
Instead of dragging and dropping certain components, with the "editor-js" you would have access to a list of components.

![image](https://user-images.githubusercontent.com/111701513/215019187-2f13bfe4-39ed-47e7-b344-90044b434c84.png)

*That is, we would have a chained block method. This allows anyone to develop their website with no programming knowledge. There are things today like nocode, lowcode - I wonder if there wouldn't be a market or a possibility for tinacms to have that. nocode, lowcode as baserow, airtable etc*
